design/rfc-01-fwu-suit/rfc-01.rst

```diff
@@ -0,0 +1,693 @@
+
+*   24: As for the earlier processing phase, the Update client uses ``psa_fwu_process()`` to process of the installation commands sequences. If this sequence requests additional payloads, this results in payload fetch requests, in the same manner as in the FETCHING state, see steps 7 - 21 in `Figure 3 <fig-fetch_>`_.
+
+*   26: After the SUIT installation has completed, the system can then load and invoke the updated firmware.
+
+Fetching during install (after reboot)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In systems where the installation occurs following reboot, or in a special execution state of the system, the implementation might choose to use the Firmware Update API between a trusted installer and a payload fetcher application. Although the interfaces used for this can be implementation-defined, `Figure 5 <fig-installer_>`_ is an example of how this could be done using the Firmware Update API.
+
+.. _fig-installer:
+
+.. figure:: installer-sequence.svg
+
+   **Figure 5** *Example use of the Firmware Update API for a dedicated Installer state*
+
+   This shows the operation of the Installer and Payload fetcher following a reboot with a STAGED envelope.
+
+This example replaces the flow in `Figure 3 <fig-fetch_>`_ from step 25 onwards.
+
+*   26: the bootloader determines that there is an update to install, and runs the Installer to process the SUIT manifest.
+
+*   27: The Payload fetcher in this example is the client of the Firmware Update API.
+
+*   28: The payload fetcher requests the processing of the SUIT manifest, and handles requests to fetch additional payloads.
+
+    The flow here is the same as the payload fetching phase that can follow the call to ``psa_fwu_install()`` for components that do not require a reboot. See steps 24 - 26 in `Figure 4 <fig-no-reboot_>`_.
+
+*   40: After the installation processing completes, the Payload fetcher returns control to the Installer, which can restart the system to finally execute the updated firmware.
+
+Errors
+~~~~~~
+
+If the Update service encounters an error during SUIT processing, the envelope component will be put into the FAILED state, and the Update client will need to use ``psa_fwu_clean()`` as usual to restore the initial firmware status.
+
+If an error occurs while transferring a payload component, that payload component will be put into a FAILED state. The Update client can remove the partial transfer using a call to ``psa_fwu_clean()`` on the payload component, and then retry the transfer, or decide to abort the entire update (see `Aborting an update`_).
+
+Dependency manifests
+~~~~~~~~~~~~~~~~~~~~
+
+In SUIT, a dependency manifest (one nested within a dependent manifest) is not processed in an isolated manner. Instead, all dependency manifests, including those nested at deeper levels, are identified and fetched before any other payload. Then the full set of manifests is involved in each of the subsequent command processing stages of the SUIT top-level (root) manifest.
```

Comment:
"Then the full set of manifests is involved...," Actually - Currently processed command sequence MAY execute 'process-dependency' on any dependencies of currently processed manifests.